### PR TITLE
ES2015ify and require Node.js 4

### PR DESCRIPTION
**Pull request by 1000ch**
Related https://github.com/imagemin/meta/issues/15 & https://github.com/imagemin/meta/issues/16.

### CODE REVIEW

File: test/test.js [hunk 1]
```diff
@@ -1,85 +1,55 @@
...
-					return;
-				}
-
-				assert(stats.isFile());
-				cb();
-			});
+		.cmd(`./configure --prefix="${tmp}"`)
+		.cmd(`make install BINPREFIX="${tmp}"`)
+		.run(err => {
+			t.true(!err);
```
Comment (sindresorhus) [inline test/test.js]: `t.ifError(err)`

File: test/test.js [hunk 2]
```diff
@@ -1,85 +1,55 @@
...
+			t.true(pathExists.sync(path.join(tmp, 'pngquant')));
```
Comment (sindresorhus) [inline test/test.js]: We can just use `fs.existsSync()`. It was undeprecated.

File: test/test.js [hunk 3]
```diff
@@ -1,85 +1,55 @@
...
+
+test.beforeEach(async () => {
+	await mkdirp(tmp);
 });
 
-afterEach(function (cb) {
-	rimraf(tmp, cb);
+test.afterEach(async () => {
+	await rimraf(tmp);
 });
```
Comment (sindresorhus) [inline test/test.js]: Would be better to use an actual temp directory, so these `beforeEach` and `afterEach` hooks wouldn't be needed. You can use: https://github.com/sindresorhus/tempy

**Comment by 1000ch** [comment]
@sindresorhus Thank you for review! Fixed.

File: test/test.js [hunk 4]
```diff
@@ -1,85 +1,48 @@
...
-
-				assert(stats.isFile());
-				cb();
-			});
+		.cmd(`./configure --prefix="${tmp}"`)
+		.cmd(`make install BINPREFIX="${tmp}"`)
+		.run(err => {
+			t.ifError(err);
+			t.true(fs.existsSync(path.join(tmp, 'pngquant')));
+			rimraf.sync(tmp);
```
Comment (sindresorhus) [inline test/test.js]: No need to do this. The system will handle cleaning up stuff in the temp directory. That's why it's called a temp directory.

File: test/test.js [hunk 5]
```diff
@@ -1,85 +1,48 @@
...
+const fs = require('fs');
+const path = require('path');
+const test = require('ava');
+const execa = require('execa');
+const tempy = require('tempy');
+const binCheck = require('bin-check');
+const BinBuild = require('bin-build');
+const compareSize = require('compare-size');
+const rimraf = require('rimraf');
+const pngquant = require('../');
```
Comment (sindresorhus) [inline test/test.js]: `../` => `..`

**Comment by sindresorhus** [comment]
Excellent :)

Can you apply these changes to your other PRs too?

**Comment by 1000ch** [comment]
@sindresorhus Applied changes to other PRs ✅

**Comment by 1000ch** [comment]
@sindresorhus Any other points or ready to merge and bump major version?

**Comment by sindresorhus** [comment]
> and bump major version?

Not really any reason to bump a new version yet. None of these changes improve anything from the user. I usually only do a new release when it has any end-user benefits, like less dependencies or newer dependencies. This PR changes nothing with the dependencies. We can bump a major version when there's some actual user-facing changes, like a new `pngquant` binary version.

**Comment by sindresorhus** [comment]
Great work @1000ch! 🙌 🎉

**Comment by sindresorhus** [comment]
I did some additional tweaks I didn't notice in the PR: https://github.com/imagemin/pngquant-bin Would you mind applying those to the other PRs as well?

**Comment by sindresorhus** [comment]
I did some additional tweaks I didn't notice in the PR: https://github.com/imagemin/pngquant-bin/commit/6ef1b8bb990be4e9de6b693927a9ceed6a8a6d4d Would you mind applying those to the other PRs as well?

**Comment by 1000ch** [comment]
@sindresorhus 👌

**Comment by 1000ch** [comment]
Applied to remaining PRs.

**Comment by sindresorhus** [comment]
Woot! Amazing stuff @1000ch :)

**Pull request merged**